### PR TITLE
Expose unwrapStore as a low level API

### LINKED ISSUE #6135: [✨] Add the `unwrapProxy` as a Low-Level API
### Is your feature request related to a problem?

As the proxy object is not cloneable, so for example, it is need to be serialized/deserialized to store into the indexedDB.

### Describe the solution you'd like

If there is a way to get the target object of the proxy object as like as the Vue's `toRaw`, we can use it directly to store into the indexedDB.

### Describe alternatives you've considered

The`getProxyTarget` function can also solve this problem.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
# What is it?

- [x] Feature / enhancement
- [ ] Bug
- [ ] Docs / tests / types / typos
- [ ] Infra

# Description
`unwrapProxy` was marked as a `@public` API in JSDoc but wasn't exposed to the user.
Exposing `unwrapProxy` enables developer to clone the content of a `useStore()` using `structureClone` or IndexedDB

fix #6135

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [x] I have ran `pnpm change` and documented my changes
- [x] I have made corresponding changes to the Qwik docs
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Review by wmertens — CHANGES_REQUESTED** [review]
I'm ok with this change, but in v2 we're calling this `unwrapStore` which is a better name IMHO.

Can you change the export to `unwrapStore`?

**Comment by wmertens** [comment]
@GrandSchtroumpf can you update the change set and API docs so they all say unwrapStore?

@shairez shall we put this in 1.9.2?

**Comment by GrandSchtroumpf** [comment]
@wmertens I updated the changeset. The doc still refers to unwrapProxy because this is the original name of the function. I can change `unwrapProxy` to `unwrapStore` everywhere, but I don't want to create conflict with v2.

File: .changeset/short-cycles-bake.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@builder.io/qwik': patch
+---
+
+Expose unwrapStore as a low level API
```
Comment (shairez) [inline .changeset/short-cycles-bake.md:5]: Please use the prefix `FEAT:` to indicate it's a feature

File: .changeset/short-cycles-bake.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@builder.io/qwik': patch
```
Comment (shairez) [inline .changeset/short-cycles-bake.md:2]: Please change this to "minor" as it's adding a feature

**Comment by shairez** [comment]
Thanks @GrandSchtroumpf !

Add a few short comments about the changesets
After that we can merge

@wmertens please resolve the "request changes" when you have a chance

**Review by wmertens — DISMISSED** [review]

**Comment by shairez** [comment]
nvm @GrandSchtroumpf 
I did the modifications on the changeset file because I wanted to merge asap to make this part of 1.9.2

But please read my review just as FYI for next time, thanks again for the PR! 🙏

**Review by shairez — APPROVED** [review]

**Pull request merged**